src/common/util/include/openvino/util/mmap_object.hpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief A header file for definition of abstraction over platform specific shared memory map objects
+ * @file mmap_object.hpp
+ */
+
+#pragma once
+
+#include <memory>
+
+#include "ngraph/runtime/aligned_buffer.hpp"
+#include "openvino/op/constant.hpp"
+
+namespace ov {
+
+std::shared_ptr<ngraph::runtime::AlignedBuffer> load_mmap_object(const std::string& path);
+
+#ifdef OPENVINO_ENABLE_UNICODE_PATH_SUPPORT
+
+std::shared_ptr<ngraph::runtime::AlignedBuffer> load_mmap_object(const std::wstring& path);
+
+#endif  // OPENVINO_ENABLE_UNICODE_PATH_SUPPORT
+
+/// \brief Creates a constant based on a passed SharedBuffer object (parent should be an instance of the SharedBuffer)
+/// with requested type/shape and offset/size in original buffer
+/// \param[in] type Type of the Constant object
+/// \param[in] shape Shape of the Constant object
+/// \param[in] parent Original instance of the SharedBuffer object (can contains several constants in one buffer)
+/// \param[in] offset Offset in the original SharedBuffer object
+/// \param[in] size Size of the Constant data in the SharedBuffer object
+/// \returns Returns a shared_ptr on a created Constant
+std::shared_ptr<ov::op::v0::Constant> get_mmap_constant(const ov::element::Type type,
+                                                        const ov::Shape shape,
+                                                        std::shared_ptr<ngraph::runtime::AlignedBuffer> parent,
```

Comment:
is it possible to avoid `ngraph`? If it is easy, please use `ov`

Reply:
yes, using fresh master